### PR TITLE
Serve Flask Prometheus Metrics; Relock Dependencies

### PR DESCRIPTION
**Pull request by KPrasch**
**Type of PR:**
Feature

**Required reviews:** 
1

**What this does:**
- Introduces https://pypi.org/project/prometheus-flask-exporter/ with a path counter
- Relock dependencies for an updated nucypher-core (`v0.10.3`) and nucypher (`v7.0.0-rc.3`)
- Indicate minimum python version as 3.9 and maximum as 3.11
- Run CI jobs on python 3.9 and 3.11

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,105 +1,103 @@
...
-mnemonic==0.20 ; python_version >= '3.5'
+mnemonic==0.20
 msgpack==1.0.5
 msgpack-python==0.5.6
-multidict==5.2.0 ; python_version >= '3.6'
-mypy-extensions==1.0.0 ; python_version >= '3.5'
+multidict==5.2.0
+mypy-extensions==1.0.0
 git+https://github.com/nucypher/nucypher.git@2b1c7d6df207ecd239db4331f452332dbcd6a8a9#egg=nucypher
 nucypher-core==0.12.0
```
Comment (KPrasch) [inline requirements.txt:56]: no updates here (yet) 👍🏻

File: Pipfile [hunk 2]
```diff
@@ -10,6 +10,7 @@ python_version = "3"
 nucypher = {git = "https://github.com/nucypher/nucypher.git", ref = "v7.0.0-rc.1"}
```
Comment (KPrasch) [inline Pipfile:10]: no updates here (yet) either 👍🏻

File: requirements.txt (hunk 1 above)
Comment (KPrasch) [reply requirements.txt]: ✔️

File: Pipfile (hunk 2 above)
Comment (KPrasch) [reply Pipfile]: ✔️

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -12,7 +12,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ["3.8", "3.10"]
+        python-version: ["3.9", "3.10"]
```
Comment (derekpierre) [inline .github/workflows/tests.yml:15]: ?
```suggestion
        python-version: ["3.9", "3.11"]
```

**Review by derekpierre — APPROVED** [review]
🎸  - See suggestion in gh action for python version.

**Pull request merged**